apps/YellowOnBoardActuators/components/outputActuator/outputActuator.c

```diff
@@ -38,64 +42,74 @@ static le_result_t WriteStringToFile(const char *path, const char *s)
     return res;
 }
 
+//--------------------------------------------------------------------------------------------------
+/**
+ * Callback that gets called when the Data Hub delivers an update to an output resource that we
+ * created.
+ */
+//--------------------------------------------------------------------------------------------------
 static void ActuatorPushHandler
 (
     double timestamp,
     bool value,
-    void *context
+    void *context   //< null-terminated sysfs path string.
 )
 {
-    struct OutputActuator *oa = context;
-    le_result_t res = WriteStringToFile(oa->sysfsPath, value ? "1" : "0");
-    if (res != LE_OK)
-    {
-        return;
-    }
+    const char* sysfsPath = context;
+
+    LE_ASSERT(LE_OK == WriteStringToFile(sysfsPath, value ? "1" : "0"));
 }
 
-le_result_t RegisterOutputActuator(const char *dhubBasePath, const char *sysfsPath)
+//--------------------------------------------------------------------------------------------------
+/**
+ * Register a Boolean actuator, making it appear at a given resource path in the Data Hub and
+ * relaying setpoints from that Data Hub resource to a given sysfs file. After calling this
+ * function, Boolean updates to the Data Hub output resource at dhubBasePath + "/enable" will
+ * result in either a '0' or a '1' being written to the file at sysfsPath.
+ *
+ * The default value will be False ('0').
+ */
+//--------------------------------------------------------------------------------------------------
+void outputActuator_Register
+(
+    const char *dhubBasePath,   //< Resource path in Data Hub's namespace for the calling app.
+    const char *sysfsPath   //< Path in file system of sysfs file to be controlled from Data Hub.
+)
 {
     /*
-     * TODO: The current design assumes that once an actuator has been registered, it can't be
-     * deregistered so I make no attempt to hang onto the pointer so that it can be cleaned up
-     * later.
+     * NOTE: The current design assumes that once an actuator has been registered, it can't be
+     * deregistered.
      */
-    struct OutputActuator *oa = calloc(sizeof(*oa), 1);
-    LE_ASSERT(oa);
-    le_result_t res = LE_OK;
-
-    strncpy(oa->sysfsPath, sysfsPath, SYSFS_PATH_LENGTH);
-    LE_FATAL_IF(oa->sysfsPath[SYSFS_PATH_LENGTH - 1] != '\0', "path too long");
 
-    int snprintfRes = snprintf(oa->dhubOutputPath, sizeof(oa->dhubOutputPath), "%s/%s", dhubBasePath, "value");
-    if (snprintfRes < 0)
-    {
-        LE_ERROR("Failed to format output path");
-        res = LE_FAULT;
-        goto error_output;
-    }
-    else if (snprintfRes >= sizeof(oa->dhubOutputPath))
+    // Construct the full Data Hub resource path (on the stack, because we won't need it once we
+    // return from this function).
+    char dhubOutputPath[DHUB_MAX_RESOURCE_PATH_LEN];
+    int len = snprintf(dhubOutputPath, sizeof(dhubOutputPath), "%s/%s", dhubBasePath, "enable");
+    if (len >= sizeof(dhubOutputPath))
```

Comment:
I'm not a big fan of LE_FATAL_IF(), to be honest.  I find it harder to read than the form you see here, and it only saves a few characters of typing.
...besides, this form increases my KLOCs. ;)

Reply:
Personally, I prefer it for testing expressions without side effects because it's more compact and reads well:
`LE_FATAL_IF(sizeof(x) > 17, "x is too big");`
vs.
```
if (sizeof(x) > 17)
{
    LE_FATAL("x is too big!");
}
```
But I don't like it for stuff like this:
`LE_FATAL_IF(SendData(destination, payload) != 17, "Wrong number of bytes sent");
I think this hides the actions of the code in what looks like a bounds check at a quick glance.

I realize that this is pretty subjective, so I don't care if you leave it as-is as.